urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/Torneo.java

```diff
@@ -0,0 +1,131 @@
+
+public class Torneo {
+
+	private int cantidadVictorias;
+	public static int numeroTorneosCreados;
+	public String datosTorneo;
+	private static final File directorio = new File("setup/tournament");
+	
+	public Torneo(int[] estrategia, int[][] estrategiasCompetidoras, int tamTablero) {
+		super();
+		leerParametros();
+		datosTorneo="";
+		numeroTorneosCreados++;
+		cantidadVictorias=0;
+		iniciarTorneo(estrategia, estrategiasCompetidoras, tamTablero);
+	}
+
+	private void iniciarTorneo(int[] estrategia1, int[][] estrategiasCompetidoras, int tamTablero) {
+		datosTorneo += "\r\n------------------TORNEO "+numeroTorneosCreados+"----------------- \r\n";
+		datosTorneo += "\r\nTotal partidas jugadas: "+estrategiasCompetidoras.length+"\r\n\r\n";
+		JuegoTorneo juego;
+		for ( int i=0 ; i<estrategiasCompetidoras.length; i++){
+			datosTorneo += "*Juego número "+(i+1)+":\r\n";
+			int[] estrategia2 = new int[4];
+			for( int j=0 ; j<estrategiasCompetidoras[i].length-1 ; j++){
+				estrategia2[j] = estrategiasCompetidoras[i][j];
+			}
+			
+			datosTorneo += "     Estrategia 1 = [ ";
+			for(int j=0;j<4;j++){
+				datosTorneo += ""+estrategia1[j];
+				if (j<3) datosTorneo += ", ";
+				else datosTorneo += " ]\r\n"; 
+			}
+			datosTorneo += "     Estrategia 2 = [ ";
+			for(int j=0;j<4;j++){
+				datosTorneo += ""+estrategia2[j];
+				if (j<3) datosTorneo += ", ";
+				else datosTorneo += " ]\r\n"; 
+			}
+			
+			//alterna el color del jugador con estrategia1 de acuerdo al número del torneo
+			if (numeroTorneosCreados%2 == 0 && numeroTorneosCreados%3 != 0) {
+				juego = new JuegoTorneo(estrategia1, estrategia2, TipoTablero.OCTOGONAL, tamTablero);
+			} else if (numeroTorneosCreados%3 == 0) {
+				juego = new JuegoTorneo(estrategia1, estrategia2, TipoTablero.PERSONALIZADO, tamTablero);
+			}  else {
+				juego = new JuegoTorneo(estrategia2, estrategia1, TipoTablero.CLASICO, tamTablero);
+			}
+			juego.start();
+			if(estrategia1 == juego.getGanador()){
+				datosTorneo += "     Ganó estrategia 1\r\n";
+				//System.out.println("Ganó estrategia 1");
+				cantidadVictorias++;
+			}else if(estrategia2 == juego.getGanador()){
+				datosTorneo += "     Ganó estrategia 2\r\n";
+				//System.out.println("Ganó estrategia 2");
+			}else if(juego.getGanador() == null){
+				datosTorneo += "No Ganó ninguan estrategia\r\n";
+				//System.out.println("No Ganó ninguan estrategia");
+			}
+		}
+		datosTorneo += "\r\n     Estrategia 1 obtuvo " + getCantidadVictorias() + " victorias de " 
+					   + estrategiasCompetidoras.length + ".\r\n\r\n";
+		guardarDatosTorneo(datosTorneo,"Torneos.txt");
+		datosTorneo = ""+numeroTorneosCreados;
+		guardarDatosTorneo(datosTorneo,"parametrosIniciales.txt");
+		
+	}
+	
+	public int getCantidadVictorias() {
+		return cantidadVictorias;
+	}
+	
+	public void guardarDatosTorneo(String datos, String nombreArchivo){
+		String nombreFichero = nombreArchivo;
+		if (!directorio.exists()) {
+            directorio.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)